pettingzoo/test/parallel_test.py

```diff
@@ -13,12 +22,15 @@ def parallel_api_test(par_env, num_cycles=10):
         assert set(obs.keys()) == (set(par_env.agents))
 
         done = {agent: False for agent in par_env.agents}
-        live_agents = par_env.agents[:]
+        live_agents = set(par_env.agents[:])
         has_finished = set()
         for i in range(num_cycles):
-            actions = {agent: space.sample() for agent, space in par_env.action_spaces.items() if agent in done and not done[agent]}
+            actions = {agent: par_env.action_space(agent).sample() for agent in par_env.agents if agent in done and not done[agent]}
```

Comment:
OPTIONAL: `dict.get()`
```suggestion
            actions = {agent: par_env.action_space(agent).sample() for agent in par_env.agents not done.get(agent, False)}
```